client/src/features/CampaignProspectsView.js

```diff
@@ -0,0 +1,72 @@
+import React, { Fragment, useState } from 'react';
+
+import { makeStyles } from '@material-ui/core/styles';
+
+import StepsTabs from '../features/StepsTabs';
+import DataTable from '../features/DataTable';
+import UserInputContainer from '../features/UserInputContainer';
+
+const useStyles = makeStyles((theme) => ({
+  prospectList: {
+    overflow: "auto",
+    width: "100%",
+    height: 590,
+    marginTop: 0,
+    [theme.breakpoints.down("lg")]: {
+      paddingLeft: 12,
+      paddingRight: 15,
+    },
+    [theme.breakpoints.down("md")]: {
+      paddingLeft: 10,
+      paddingRight: 10,
+    },
+    [theme.breakpoints.down("sm")]: {
+      paddingLeft: 5,
+      paddingRight: 5,
+    }
+  },
+}))
+
+const CampaignProspectsView = (props) => {
+  let { steps, data, handleClickOnAllRows, handleClickOnRow, selectedProspects } = props;
+  const classes = useStyles();
+  const [selected, setSelected] = useState(0);
+  const [value, setValue] = useState(0);
+  const [currentStepId, setCurrentStepId] = useState(0);
+
+  const handleCurrentStepId = (idx, step) => {
+    if (idx === 0) {
+      setSelected(0);
+      setCurrentStepId(0)
+    } else {
+      setSelected(idx);
+      setCurrentStepId(step.id)
+    }
+  }
+
+  data = data.filter(prospect => prospect.steps[currentStepId])
```

Comment:
```
prospect.steps[currentStepId]
```
doesn't steps[currentStepId] gets the step of that index? and if currentStepId is big, it might cause indexOfRange exception?